### PR TITLE
State changer

### PR DESCRIPTION
**Pull request by hewonjeong**
- detecting and removing expired items in item get API
- add expired mock item
- modify mock item's end time
- validChecker func in ItemManager
- item-model test code

### CODE REVIEW

File: src/server/database-mock-data.js [hunk 1]
```diff
@@ -35,7 +39,7 @@ export const mockItems = {
     modifiedDate: '2016-10-08T01:11:46.851Z',
     category: 'facility',
     startTime: '2016-10-08T01:11:46.851Z',
-    endTime: '2016-10-20T01:11:46.851Z',
+    endTime: new Date(initTime.getTime() + dayInMs * 3).toISOString(),
```
Comment (festinalente91) [inline src/server/database-mock-data.js]: facility has no endTime & startTime

File: src/server/database-mock-data.js [hunk 2]
```diff
@@ -48,7 +52,7 @@ export const mockItems = {
     modifiedDate: '2016-10-09T01:11:46.851Z',
     category: 'facility',
     startTime: '2016-10-09T01:11:46.851Z',
-    endTime: '2016-10-19T01:11:46.851Z',
+    endTime: new Date(initTime.getTime() + dayInMs * 4).toISOString(),
```
Comment (festinalente91) [inline src/server/database-mock-data.js]: ditto

File: src/server/items/models.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
+import db from '../database';
+import {KeyUtils, STATE, DEFAULT_PRECISON} from '../key-utils';
+
+export default class ItemManager {
```
Comment (festinalente91) [inline src/server/items/models.js:4]: I think this belongs to key-utils.js.
this state change function can be shared the other entities(images, reports, etc ...)  later.

File: src/server/key-utils.js [hunk 4]
```diff
@@ -76,10 +76,10 @@ export const KeyUtils = {
     });
     return `${key}-${timeHash}`;
   },
-  getIdxKeys: (lat, lng, timeHash) => {
+  getIdxKeys: (lat, lng, timeHash, state = STATE.ALIVE) => {
```
Comment (festinalente91) [inline src/server/key-utils.js:79]: cool~

File: src/test/node/items/models.js [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+  createdDate: '2016-10-13T01:11:46.851Z',
+  modifiedDate: '2016-10-13T01:11:46.851Z',
+  category: 'event',
+  startTime: '2016-10-13T01:11:46.851Z',
+  endTime: '2016-10-15T01:11:46.851Z',
+  key: 'item-8523910540005-edddd-473b-1234-ead0f190b000',
+  userKey: 'user-8523574664000-b82e-473b-1234-ead0f54gvr00',
+  image: 'aaaaaa'
+};
+test('distinguish expired date', t => {
```
Comment (ovekyc) [inline src/test/node/items/models.js]: how about 'validate expired date' ?

File: src/test/node/items/models.js [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+      t.pass('all item keys are changed successfully');
+      t.equal(idxItems.length, expected.numberOfIdxItems,
+      'should be same number of indexing items');
+      t.end();
+    });
+  })
+  .catch((err)=>{
+    /* eslint-disable no-console */
+    console.log(err);
+    /* eslint-enable */
```
Comment (ovekyc) [inline src/test/node/items/models.js]: t.fail()
t.end()

File: src/test/node/items/models.js [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+  });
+});
+test('Check endTime value and chang indexing items', t => {
+  const expected = {
+    result: false,
+    numberOfIdxItems: DEFAULT_PRECISON
+  };
+  const expiredItemKey = 'item-8523910540006-b82e-473b-1234-ead0f190b006';
```
Comment (ovekyc) [inline src/test/node/items/models.js]: please import from mockItems

File: src/test/node/items/models.js [hunk 8]
```diff
@@ -0,0 +1,128 @@
...
+  .then((changedItemsCnt)=>{
+    t.equal(changedItemsCnt, expected.numberOfIdxItems,
+     'should be same number of indexing items');
+    t.pass('all item keys are changed successfully');
+    t.end();
+  })
+  .catch((err)=>{
+    /* eslint-disable no-console */
+    console.log(err);
+    /* eslint-enable */
```
Comment (ovekyc) [inline src/test/node/items/models.js]: ditto

File: src/server/items/models.js (hunk 3 above)
Comment (hewonjeong) [reply src/server/items/models.js:4]: OK I will move it to key-utils in next branch

File: src/test/node/items/models.js (hunk 5 above)
Comment (hewonjeong) [reply src/test/node/items/models.js]: good!

File: src/server/items/models.js [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+  static validChecker(item, cb) {
```
Comment (festinalente91) [inline src/server/items/models.js:5]: it seems more like an expiredChecker

File: src/server/items/models.js [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+            {type: 'del', key: idxKeys[i]}
+          );
+        }
+      }
+    }
+    db.batch(ops, (err) => {
+      return (err) ? cb(err) : cb();
+    });
+  }
+  static isValid(endTime) {
```
Comment (festinalente91) [inline src/server/items/models.js:31]: what about isExpired ?

File: src/server/items/models.js (hunk 3 above)
Comment (festinalente91) [reply src/server/items/models.js:4]: nice

File: src/test/node/items/models.js (hunk 7 above)
Comment (hewonjeong) [reply src/test/node/items/models.js]: resolve!

File: src/server/items/models.js (hunk 9 above)
Comment (hewonjeong) [reply src/server/items/models.js:5]: I think 'isValid' == true -> send to user is more easy to understand.

File: src/server/items/models.js (hunk 10 above)
Comment (hewonjeong) [reply src/server/items/models.js:31]: same here.

**Comment by festinalente91** [comment]
LGTM

**Comment by ovekyc** [comment]
LGTM

**Pull request merged**